tests/bgp/test_bgp_azng_migration.py

```diff
@@ -105,89 +134,379 @@ def test_bgp_azng_migration(duthosts, enum_upstream_dut_hostname):
+                "Check BGP session status, route advertisement policies, and the output of the command.\n"
+                "- Command: {}\n"
+                "- Output: {}\n"
+            ).format(
+                bgp_nbr_recv_cmd if 'bgp_nbr_recv_cmd' in locals() else 'N/A',
+                routes_json
+            )
+
+            assert routes_json['filteredPrefixCounter'] == 0, (
+                "Filtered prefix counter is not zero after AZNG migration rollback. "
+                "Expected filteredPrefixCounter == 0, but got {}. "
+                "- Command: {}\n"
+            ).format(
+                routes_json['filteredPrefixCounter'],
+                bgp_nbr_adv_cmd
+            )
 
             if "ipv6" in bgp_nbr_adv_cmd:
-                assert routes_json['totalPrefixCounter'] == original_ipv6_route_adv_count
-                assert routes_json['filteredPrefixCounter'] == original_ipv6_route_adv_filter_count
+                assert routes_json['totalPrefixCounter'] == original_ipv6_route_adv_count, (
+                    "Mismatch in total advertised IPv6 route count after AZNG migration. "
+                    "Expected: {}, Actual: {}. "
+                    "- Command: {}\n"
+                ).format(
+                    original_ipv6_route_adv_count,
+                    routes_json['totalPrefixCounter'],
+                    bgp_nbr_adv_cmd
+                )
+                assert routes_json['filteredPrefixCounter'] == original_ipv6_route_adv_filter_count, (
+                    "Mismatch in filtered advertised IPv6 route count after AZNG migration. "
+                    "Expected: {}, Actual: {}. "
+                    "- Command: {}\n"
+                ).format(
+                    original_ipv6_route_adv_filter_count,
+                    routes_json['filteredPrefixCounter'],
+                    bgp_nbr_adv_cmd
+                )
             else:
-                assert routes_json['totalPrefixCounter'] == original_ipv4_route_adv_count
-                assert routes_json['filteredPrefixCounter'] == original_ipv4_route_adv_filter_count
+                assert routes_json['totalPrefixCounter'] == original_ipv4_route_adv_count, (
+                    "Mismatch in total advertised IPv4 route count after AZNG migration. "
+                    "Expected: {}, Actual: {}. "
+                    "- Command: {}\n"
+                ).format(
+                  original_ipv4_route_adv_count,
+                  routes_json['totalPrefixCounter'],
+                  bgp_nbr_adv_cmd
+                )
+                assert routes_json['filteredPrefixCounter'] == original_ipv4_route_adv_filter_count, (
+                   "Mismatch in filtered advertised IPv4 route count after AZNG migration. "
+                   "Expected: {expected}, Actual: {actual}. "
+                   "- Command: {cmd}\n"
+                ).format(
+                  expected=original_ipv4_route_adv_filter_count,
```

Comment:
why do you use another format to print the variables, please keep the format same consistently.